addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -77,59 +77,81 @@ function factory(dependencies) {
                     return;
                 }
                 const [, model, id] = channel;
-                switch (model) {
-                    case 'ir.needaction':
-                        return this._handleNotificationNeedaction(message);
-                    case 'mail.channel':
-                        return this._handleNotificationChannel(id, message);
-                    case 'res.partner':
+                switch (message.type) {
+                    case 'channel_seen':
+                        return this._handleNotificationChannelSeen(message.payload);
+                    case 'channel_fetched':
+                        return this._handleNotificationChannelFetched(message.payload);
+                    case 'channel_new_message':
+                        return this._handleNotificationChannelMessage(message.payload);
+                    case 'deletion':
+                        return this._handleNotificationPartnerDeletion(message.payload);
+                    case 'inbox_needaction':
+                        return this._handleNotificationNeedaction(message.payload);
+                    case 'mark_as_read':
+                        return this._handleNotificationPartnerMarkAsRead(message.payload);
+                    case 'message_notification_update':
+                        return this._handleNotificationPartnerMessageNotificationUpdate(message.payload);
+                    case 'simple_notification':
+                        return this._handleSimpleNotification(message.payload);
+                    case 'toggle_star':
+                        return this._handleNotificationPartnerToggleStar(message.payload);
+                    case 'transient_message':
+                        return this._handleNotificationPartnerTransientMessage(message.payload);
+                    case 'typing_status':
+                        return this._handleNotificationChannelTypingStatus(message.payload);
+                    case 'unsubscribe':
                         if (id !== this.env.messaging.currentPartner.id) {
                             // ignore broadcast to other partners
                             return;
                         }
-                        return this._handleNotificationPartner(Object.assign({}, message));
+                        return this._handleNotificationPartnerUnsubscribe(message.payload);
+                    case 'author':
+                        if (id !== this.env.messaging.currentPartner.id) {
+                            // ignore broadcast to other partners
```

Comment:
This needs to go, we only added it for tests, due to how the old system worked.
Basically in test we directly push in the bus in JS, but the subscription mechanism to some channels (in particular res.partner) is actually done in python as you have seen.
I guess the proper fix is to move the subscription mechanism in JS, but keep a validation in python for security reasons.
Or if we want to keep the current way, the longpolling route should be better integrated in the mock server to handle this subscription step.

I guess this comment is for a follow up PR.